// containers/admin/index.js
import React, {Component} from 'react';
import {connect} from 'react-redux'

import {
    addPhoneToShop,
    removePhoneFromShop,
    changePhoneValue
} from 'actions'

import {getPhones1} from 'selectors'
import * as R from "ramda";
import {Link} from "react-router";

class Admin extends Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
        let initialState = {price: -1};
        this.state = {phonesData: initialState};
        window.location.assign("?#");
    }


    handleChange(event) {
        event.preventDefault();
        if (event.target.value < 0 ) {
            alert("Цена меньше нуля!");
            setTimeout(() => window.location.assign("#/"), 1000);
        }
        let id = event.target.id;
        let newState = Object.assign({}, this.state.phonesData);
        newState[id] = event.target.value;
        this.setState({phonesData: newState});
        if (event.toLocaleString())
        setTimeout(() => window.location.assign("#/"), 1000);
    }

    render() {
        const {phones} = this.props;
        const {addPhoneToShop} = this.props;
        const {removePhoneFromShop} = this.props;
        const {changePhoneValue} = this.props;
        console.log("phones = ", this.props);

        return (
            <div>
                <div className='books row'>
                    {phones.map((phone, index) => (
                        <form onSubmit={() => changePhoneValue(phone.id, this.state.phonesData.price)}>
                        <tr
                            key={index}
                            className='item-checout'
                        >
                            <td className='first-column-checkout'>
                                <img
                                    className='img-thumbnail'
                                    src={phone.image}
                                    alt={phone.name}
                                />
                            </td>
                            <td>{phone.name}</td>
                            <td>
                                ${phone.price}
                            </td>
                            <td>Новая цена: $
                                <input type="number" id = 'price' required="true"
                                       onChange={(event) => this.handleChange(event)}
                                />
                            </td>
                            <td>
                                <button type="submit">Подтвердить новую цену</button>
                            </td>
                            <td>{phone.count}</td>
                            <td>
                                   <span
                                       className='delete-cart'
                                       onClick={() => removePhoneFromShop(phone.id)}
                                   />
                            </td>
                            <td>
                                <button onClick={() => addPhoneToShop(phone.id, phone.count)}>Добавить</button>
                            </td>
                        </tr>
                        </form>
                    ))}
                </div>
            </div>

        )
    }


    renderPhone(phone, index) {
        const shortDescription = `${R.take(60, phone.description)}...`

        return (
            <div className='col-sm-4 col-lg-4 col-md-4 book-list' key={index}>
                <div className='thumbnail'>
                    <img
                        className='img-thumbnail'
                        src={phone.image}
                        alt={phone.name}
                    />
                    <div className='caption'>
                        <h4 className='pull-right'>${phone.price}</h4>
                        <h4>
                            <Link to={`/phones/${phone.id}`}>
                                {phone.name}
                            </Link>
                        </h4>
                        <h4 className='amount'>Amount: {phone.count}</h4>
                        <p>{shortDescription}</p>
                    </div>
                </div>
            </div>
        )
    }


}

const mapStateToProps = (state, ownProps) => {
    return {
       phones: state.admin
    }
};

const mapDispatchToProps = {
    addPhoneToShop,
    removePhoneFromShop,
    changePhoneValue
};

export default connect(mapStateToProps, mapDispatchToProps)(Admin)
